File: 05-reflection-ii/java/UseLogger.java
class Info {
	private final int num; 
	private String name;
	
	public Info(int num, String name) {
		this.num = num;
		this.name = name;
	}
	
	public int getNum() { return num; }

	public String getName() { return name; }
	public void setName(String value) { name = value; }
}

public class UseLogger {
	public static void main(String[] args) throws Exception {
		Logger.log(null);
		Logger.log("ISEL");
		Logger.log(new Info(12345, "Fernando Pessoa"));
		Logger.log(new Info(23456, "Afonso Henriques"));
		
		final var hmap = new java.util.HashMap<Integer, String>();
		hmap.put(12345, "Fernando Pessoa");
		hmap.put(23456, "Afonso Henriques");
		
		Logger.log(hmap);
	}
}

File: 05-reflection-ii/kotlin/useLogger.kt
@AltName("Data")
data class Info(val num : Int, var name : String)

data class User(val username : String, @NoLog var password : String)

fun main() {
	Logger.log(null)
	Logger.log("ISEL")
	Logger.log(Info(12345, "Fernando Pessoa"))
	Logger.log(Info(23456, "Afonso Henriques"))
	
	Logger.log(User("fpessoa", "pessoa1888"))
	
	val hmap = java.util.HashMap<Int, String>()
	hmap.put(12345, "Fernando Pessoa")
	hmap.put(23456, "Afonso Henriques")
	
	Logger.log(hmap)
}
